dvc/ignore.py

```diff
@@ -27,12 +17,48 @@ def __call__(self, root, dirs, files):
         raise NotImplementedError
 
 
-class DvcIgnoreFromFile(DvcIgnore):
-    def __init__(self, ignore_file_path, ignore_handler):
+# def full_path_pattern(pattern, path):
+#     # NOTE: '/' is translated to ^ for .gitignore style patterns,
+#     # it is natural to proceed absolute path with this sign
+#     if pattern.startswith("//"):
+#         return pattern
+#
+#     negation = False
+#
+#     if pattern.startswith("!"):
+#         pattern = pattern[1:]
+#         negation = True
+#
+#     if pattern.startswith("/"):
+#         pattern = os.path.normpath(path) + pattern
+#     else:
+#         pattern = os.path.join(path, pattern)
+#
+#     pattern = os.path.join(path, pattern)
+#     pattern = "/" + pattern
+#
+#     if negation:
+#         pattern = "!" + pattern
+#
+#     return pattern
+
+
+class DvcIgnorePatterns(DvcIgnore):
+    def __init__(self, ignore_file_path, patterns):
         self.ignore_file_path = ignore_file_path
         self.dirname = os.path.normpath(os.path.dirname(ignore_file_path))
 
-        self.ignore_spec = ignore_handler.read_patterns(ignore_file_path)
+        abs_patterns = []
+        rel_patterns = []
+        for p in patterns:
+            # NOTE: '/' is translated to ^ for .gitignore style patterns,
+            # it is natural to proceed absolute path with this sign
```

Comment:
Does gitingore support ignoring abs paths? Didn't we agree that external deps and outs should not be affected by dvcignore, so we don't need to support abs paths in `dvcignore`s?

Reply:
Oh, I must have misunderstood.
No AFAIK git does not support abs paths.